### DISCUSSION TITLE
[Question] Global state for deeply nested machines

### DISCUSSION THREAD

**Discussion by iamzapata** [statelyai/xstate#4790] [Q&A]
I'm trying to figure out how to manage global state in an existing application that has many many machines, with multiple levels of nesting.

To put this into more context, lets say I have an app with with the following structure: 

```mermaid
graph TD
A --> B
A --> C
B --> D
C --> E
D --> F
D --> G
G --> H
```

Machine `A`  loads some data, and this data is needed by machines further below. 

### Problem

In the current state, if `H` needs a business object that is inside the context of `A`, it is passed all the way down, through `B`, `D`,  and `G`. 

The machines in lower levels have valid use case for the top context, but I want to refactor the way they access it and not have to context drill. 

I'm thinking about using a light store like jotai or zustand, something outside of react, to have the root machien manage global state that is exposed globally, accessible anywhere basically. Has anyone gone down such a path?

**Comment by tivac** [comment]
We do something similar using svelte stores, it's been very effective for us. `context` is nice because it's so well integrated but becomes quite unwieldy when you start nesting machines more than a level deep!
